File: Numeros Pseudo-Aleatorios/Park Miller/ParkMiller.pas
{ 
 PRNG .......: Park-Miller
 Ano ........: 1988
 Autor ......: S. K. Park & K. W. Miller
}

program ParkMillerPRNG;

uses
  SysUtils, DateUtils;

function ParkMiller(var X: LongWord): LongWord;
begin
  X := (16807 * X) mod 2147483647;
  ParkMiller := X;
end;

var
  Xn1, Xn2: LongWord;
  i: Integer;
  
begin

  Xn1 := 1234; 
  Xn2 := LongWord(DateTimeToUnix(Now)); 

  writeln('GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)');
  writeln('PSEUDORANDOM NUMBER GENERATOR (PRNG)');
  writeln('PADRAO: Park-Miller');
  writeln('==========================================');
  writeln;
  writeln('    Semente: 1234        Semente: TIME');
  writeln('    -------------        -------------');

  for i := 1 to 10 do
    writeln(ParkMiller(Xn1):17, '        ', ParkMiller(Xn2):13);

end.

File: Numeros Pseudo-Aleatorios/Park Miller/ParkMiller.java
/**
 * PRNG .......: Park-Miller
 * Ano ........: 1988
 * Autor ......: S. K. Park & K. W. Miller
*/

import java.time.Instant;

public class Main {

    public static int ParkMiller(int X) {
        X = (16807 * X) % 2147483647;
        return Math.abs(X);
    }

    public static void main(String[] args) {
      
        int Xn1, Xn2;
        int i;

        Xn1 = 1234;
        Xn2 = (int) Instant.now().getEpochSecond();

        System.out.println("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)");
        System.out.println("PSEUDORANDOM NUMBER GENERATOR (PRNG)");
        System.out.println("PADRAO: Park-Miller");
        System.out.println("==========================================");
        System.out.println();
        System.out.println("    Semente: 1234        Semente: TIME");
        System.out.println("    -------------        -------------");

        for (i = 1; i <= 10; i++) {
            Xn1 = ParkMiller(Xn1);
            Xn2 = ParkMiller(Xn2);
            System.out.printf("%17d        %13d%n", Xn1, Xn2);
        }
      
    }
  
}
